app/controllers/translations_controller.rb

```diff
@@ -0,0 +1,14 @@
+class TranslationsController < ApplicationController
+
+  def create
+    @video = Video.find(params[:video_id])
+    begin
+      @translation = Translation.create_or_update_translation(@video, current_user, params[:srt])
+    rescue Exception => exception
+      add_flash(:alert, exception.message)
+      redirect_to video_path(@video)
+      return
+    end
+    redirect_to video_translation_path(:video_id => @video.id, :id => @translation.id)
+  end
```

Comment:
I think the way we've agreed to approach this is that creating a translation is equivalent to assigning a video to a user. That design means that the create method doesn't accept a translation file.

Instead, consider adding another method, something like `upload_srt`.